### DISCUSSION TITLE
Allow null-coalescing operator on void function

### DISCUSSION THREAD

**Discussion by Thaina** [dotnet/csharplang#2735] [Language Ideas]
Since C# could shorthand many things as one line expression. But sometimes we just want to call some void function in the expression and return value. Which is not possible

So I think it would be easier if we can write it like this

```C#
int i = IsTrue switch {
    false => Console.WriteLine("false is 0") ?? 0;
    true => Console.WriteLine("true is 1") ?? 1;
};

if(i > 0 && (Console.WriteLine("i is true, is i valid ?" + i) ?? i < 2))
    DoSomething();

var values = dictionary.Select((item) => Console.WriteLine(item.Key + " : " + item.Value) ?? item.Value).ToArray();
```

**Comment by svick** [comment]
I don't think this syntax makes sense, because `void` is not `null`. The syntax proposed by Sequence Expressions https://github.com/dotnet/csharplang/issues/377 achieves the same goal and I find it much more sensible
